scripts/apps/authoring/authoring/index.ts

```diff
@@ -12,6 +12,24 @@ import {ArticleUrlFields} from './article-url-fields';
 import {PopulateAuthorsController} from './controllers/PopulateAuthorsController';
 
 import {gettext} from 'core/utils';
+import {IArticle} from 'superdesk-interfaces/Article';
+import {IArticleSchema} from 'superdesk-interfaces/ArticleSchema';
+
+export interface IOnChangeParams {
+    item: IArticle;
+    original: IArticle;
+}
+
+export interface IArticleSchemaParams {
+    item: IArticle;
+    schema: IArticleSchema;
+}
+
+type IOnChangeMiddleware = (params: IOnChangeParams) => any;
```

Comment:
Should these two function types return a promise?
`(params: IOnChangeParams) => Promise<any>`
`(params: IArticleSchemaParams) => Promise<any>`

Reply:
I'm not sure yet how to pass that return value and keep the applyMiddleware code generic at the same time, will check it